### PR TITLE
helpers,osutil,many: move helpers to better packages

### PR DESCRIPTION
**Pull request by mvo5**
First pass of moving stuff out of the overly generic helpers package.  It also removes some unused code.

### CODE REVIEW

File: coreconfig/config.go [hunk 1]
```diff
@@ -318,7 +319,7 @@ var getTimezone = func() (timezone string, err error) {
 // setTimezone sets the specified timezone for the system, an error is returned
 // if it can't.
 var setTimezone = func(timezone string) error {
-	if err := helpers.CopyFile(filepath.Join(tzZoneInfoPath, timezone), tzZoneInfoTarget, helpers.CopyFlagOverwrite); err != nil {
+	if err := cp.CopyFile(filepath.Join(tzZoneInfoPath, timezone), tzZoneInfoTarget, cp.CopyFlagOverwrite); err != nil {
```
Comment (niemeyer) [inline coreconfig/config.go]: cp seems slightly on the opposite end: too specific for being a package. It'll be hard for us to find anything else quite fitting here. I suggest having this inside osutil for the time being, with a plan B of moving it into "fsutil" if we end up having too many filesystem manipulation routines inside osutil.

**Comment by niemeyer** [comment]
LGTM, with one suggestion.

**Comment by mvo5** [comment]
retest this please

File: cmp/cmp.go [hunk 2]
```diff
@@ -17,13 +17,15 @@
  *
  */
 
-package helpers
+package cmp
```
Comment (niemeyer) [inline cmp/cmp.go]: Aren't both of the functions here good for osutil too?

Comment (mvo5) [inline cmp/cmp.go]: Makes sense, happy to move them over.

**Comment by mvo5** [comment]
The cmp package is now merged into osutil.

**Comment by niemeyer** [comment]
@chipaca?  @pedronis?

File: osutil/chdir.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package osutil
+
+import (
+	"os"
+)
+
+// ChDir runs runs "f" inside the given directory
+func ChDir(newDir string, f func() error) (err error) {
```
Comment (chipaca) [inline osutil/chdir.go]: This one makes me nervous, as it'll only work as expected in single-threaded usage. Can we put a lock on it?

Comment (chipaca) [inline osutil/chdir.go]: And yes, I know this is not related to the branch -- putting a warning in a comment is probably good enough for now :-)

**Comment by chipaca** [comment]
:+1:

**Pull request merged**